programs/conditional_vault/src/instructions/merge_tokens.rs

```diff
@@ -48,23 +78,44 @@ impl<'info, 'c: 'info> InteractWithVault<'info> {
             )?;
         }
 
+        // Validate token accounts and decimals
+        require_eq!(
+            accs.user_underlying_token_account.mint,
+            accs.vault_underlying_token_mint.key(),
+            VaultError::UnderlyingTokenMintMismatch
+        );
+        require_eq!(
+            accs.vault_underlying_token_account.mint,
+            accs.vault_underlying_token_mint.key(),
+            VaultError::UnderlyingTokenMintMismatch
+        );
+        require_eq!(
+            accs.vault_underlying_token_mint.decimals,
+            accs.vault.decimals,
+            VaultError::AssertFailed
+        );
+
         // Transfer `amount` from vault to user
-        token::transfer(
+        token::transfer_checked(
             CpiContext::new_with_signer(
                 accs.token_program.to_account_info(),
-                Transfer {
+                TransferChecked {
```

Comment:
What's the point of `TransferChecked` if we already know that the mints are the same?

Reply:
https://github.com/metaDAOproject/futarchy/pull/254#discussion_r1830379500

I replied to it above. But if you're already sure about the mint and decimals, then you can use Transfer.